mycpp/mylib_leaky_test.cc

```diff
@@ -519,6 +519,136 @@ TEST test_sizeof() {
   PASS();
 }
 
+TEST test_str_slice() {
+  printf("\n");
+
+  Str* s0 = new Str("abcdef");
+
+  printf("------- Happy Path -------\n");
+
+  { // Happy path
+    Str* s1 = s0->slice(0, 5);
+    ASSERT(str_equals(s1, new Str("abcde")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(1, 5);
+    ASSERT(str_equals(s1, new Str("bcde")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(0, 0);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(0, 6);
+    ASSERT(str_equals(s1, new Str("abcdef")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(-6, 6);
+    ASSERT(str_equals(s1, new Str("abcdef")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(0, -6);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+  {
+    Str* s1 = s0->slice(-6, -6);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(5, 6);
+    ASSERT(str_equals(s1, new Str("f")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(6, 6);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+
+  printf("---- Infinite Sadness ----\n");
+
+  {
+    Str* s1 = s0->slice(0, -7);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(-7, -7);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(-7, 0);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(6, 6);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(7, 7);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(6, 5);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(7, 5);
+    ASSERT(str_equals(s1, new Str("")));
+    printf("%s\n", s1->data());
+  }
+
+  {
+    Str* s1 = s0->slice(7, 6);
```

Comment:
Some of these test cases seem very similar... I think  cases like  (-100 100 ) and (0 100)  actually add a bit of coverage?   there are many branches in this function

(I just got some code coverage tools working, I hope to put them in the CI)